packages/eslint-plugin/src/rules/prefer-nullish-coalescing.ts

```diff
@@ -428,22 +438,21 @@ export default createRule<Options, MessageIds>({
             } else if (isUndefinedIdentifier(testNode)) {
               hasUndefinedCheckWithoutTruthinessCheck = true;
             } else if (
-              (operator === '!==' || operator === '!=') &&
-              isNodeEqual(testNode, node.consequent)
-            ) {
-              identifierOrMemberExpression = testNode;
-            } else if (
-              (operator === '===' || operator === '==') &&
-              isNodeEqual(testNode, node.alternate)
+              isNodeOrNodeExpressionEqual(
+                testNode,
+                getNonNullishBranchNode(node, operator),
+              )
             ) {
-              identifierOrMemberExpression = testNode;
+              if (!nullishCoalescingLeftNode) {
+                nullishCoalescingLeftNode = testNode;
+              }
```

Comment:
Ironically, this line could use the nullish coalescing operator 😁 

Also the reason that it needs to be only assigned once is extremely subtle IMO and deserves a comment. WDYT?

```suggestion
                // We want only the first expression in a multi-part nullish check, 
                // since subsequent expressions might not need all the ?.s. 
                // Eg a?.b?.c !== undefined && a.b.c !== null ? a.b.c : 'foo';
                nullishCoalescingLeftNode ??= testNode;
```